### DISCUSSION TITLE
How to go about testing an SQS consumer application with rollouts

### DISCUSSION THREAD

**Discussion by xargsgrep** [argoproj/argo-rollouts#3453] [Q&A]
I have application that dequeues from SQS and doesn't serve any http traffic. I also have some cluster tests for this application which I want to be able to run in the staging environment. Since it does not serve any http traffic, it seems like the canary or blue/green strategies will not really help. I want to be able to deploy the new version of the application, shut down the old pods, and then run the cluster tests to ensure that only the new version is dequeuing. If the tests pass then it should keep the new pods, but if they fail then I want it to rollback to the old pods. How would I accomplish this workflow with argo rollouts? Or would I not be able to use rollouts at all for this?

**Comment by kostis-codefresh** [comment]
I have written a guide for this exact scenario here https://codefresh.io/blog/progressive-delivery-for-stateful-services-using-argo-rollouts/
It uses RabbitMQ but the same concepts apply to SQS as well
